clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1862,8 +1814,9 @@ void FunctionVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV,
     return;
   }
   if (this->numParams() == 0) {
-    // This is an untyped declaration, we need to perform a transplant.
-    From->brainTransplant(this, I);
+    // This is an untyped declaration, we need to switch to the other
+    assert(false && "This merge should happen in reverse");
```

Comment:
Why shouldn't this happen?

Reply:
It shouldn't be allowed because it's backwards, and without the linked pointers of brantransplant, it will not change the `this` constraint, which is the one saved.

It shouldn't occur because we check number of parameters in almost all code that reaches here. The exception is function pointers, which I was going to ask about and leave as TODO because it's probably rare code (a parameterless FP which is later redeclared with params). The fix wouldn't be hard, but would involve finding an example and adding code to preprocess `FV` merges.